pkg/pillar/types/locationconsts.go

```diff
@@ -25,8 +25,10 @@ const (
 	VolumeClearDirName = ClearDirName + "/volumes"
 	// PersistDebugDir - Location for service specific debug/traces
 	PersistDebugDir = PersistDir + "/agentdebug"
-	//VolumeZFSPool - pool for create volumes
-	VolumeZFSPool = "persist" + "/volumes"
+	//VolumeClearZFSDataset - dataset to create volumes without encryption
+	VolumeClearZFSDataset = "persist" + "/clear" + "/volumes"
+	//VolumeEncryptedZFSDataset - dataset to create volumes with encryption
+	VolumeEncryptedZFSDataset = "persist" + "/vault" + "/volumes"
```

Comment:
This seems to pretty error prone -- I suggest you reconcile it with VolumeEncryptedDirName and VolumeClearDirName somehow

Reply:
I moved them to depend on *DirName.